src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/StreamExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.Shared.Diagnostics;
+
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Provides extension methods for <see cref="Stream"/>.</summary>
+public static class StreamExtensions
+{
+    /// <summary>Converts a <see cref="Stream"/> to an <see cref="IAsyncEnumerable{DataContent}"/>.</summary>
+    /// <param name="stream">The <see cref="Stream"/> to convert.</param>
+    /// <param name="mediaType">The optional media type of the audio stream.</param>
+    /// <param name="bufferSize">The optional buffer size to use when reading from the audio stream. The default is 4096.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>An <see cref="IAsyncEnumerable{DataContent}"/>.</returns>
+    public static async IAsyncEnumerable<DataContent> ToAsyncEnumerable(
```

Comment:
Can we delete this for now? It doesn't appear to be used in this PR, and if it's necessary for another provider, it can be done in that provider until it's actually proved to be valuable enough to include.

Reply:
Yeah, I was intending to remove it, missed after removing the `IAsyncEnumerable` from API's